R/CallbackTuning.R

```diff
@@ -0,0 +1,135 @@
+#' @title Create Tuning Callback
+#'
+#' @description
+#' Specialized [bbotk::CallbackOptimization] for tuning.
+#' Callbacks allow to customize the behavior of processes in mlr3tuning.
+#' The [callback_tuning()] function creates a [CallbackTuning].
+#' Predefined callbacks are stored in the [dictionary][mlr3misc::Dictionary] [mlr_callbacks] and can be retrieved with [clbk()].
+#' For more information on tuning callbacks see [callback_tuning()].
+#'
+#' @examples
+#' # write archive to disk
+#' callback_tuning("mlr3tuning.backup",
+#'   on_optimization_end = function(callback, context) {
```

Comment:
I think I would now prefer to call the argument `callback` `self`. 
I think this makes it easier for people who are familar with R6 / python class system and people who are not will not be able to implement it anyway.

Reply:
I would rather assume that most people have no idea about R6 and that it is clearer that way.